tools/repack-kernel

```diff
@@ -0,0 +1,235 @@
+            ubuntu-core-initramfs create-initrd \
+                                  --kernelver "$kver" \
+                                  --skeleton "$skeletondir" \
+                                  --feature main \
+                                  --kerneldir "lib/modules/$kver" \
+                                  --output "$target/work/repacked-initrd"
+        )
+
+        # assumes all files are named <name>-$kver
+        ubuntu-core-initramfs create-efi \
+                              --kernelver "$kver" \
+                              --initrd repacked-initrd \
+                              --kernel vmlinuz \
+                              --output repacked-kernel.efi
+
+        cp "repacked-kernel.efi-$kver" "$target/kernel/kernel.efi"
+
+        # XXX: needed?
+        chmod +x "$target/kernel/kernel.efi"
+    )
+}
+
+cull_firmware() {
+    local target="$1"
+
+    if [ -z "$target" ]; then
+        echo "usage: cull-firmware <extract-tree>"
+        exit 1
+    fi
+    (
+        # XXX: drop ~450MB+ of firmware which should not be needed in under qemu
+        # or the cloud system
+        cd "$target/kernel"
+        rm -rf firmware/*
+    )
+}
+
+cull_modules() {
+    local target="$1"
+
+    if [ -z "$target" ]; then
+        echo "repack-kernel: missing target for cull-modules"
+        exit 1
+    fi
+
+    target="$(realpath $target)"
+    kver="$(get_kver "$target/kernel")"
+
+    (
+        cd "$target/kernel"
+        # drop unnecessary modules
+        awk '{print $1}' <  /proc/modules  | sort > "$target/work/current-modules"
+        #shellcheck disable=SC2044
+        for m in $(find modules/ -name '*.ko'); do
+            noko=$(basename "$m"); noko="${noko%.ko}"
+            if echo "$noko" | grep -f "$target/work/current-modules" -q ; then
+                echo "keeping $m - $noko"
+            else
+                rm -f "$m"
+            fi
+        done
+
+        # depmod assumes that /lib/modules/$kver is under basepath
+        mkdir -p fake/lib
+        ln -s "$PWD/modules" fake/lib/modules
+        depmod -b "$PWD/fake" -A -v "$kver"
+        rm -rf fake
+    )
+}
+
+pack() {
+    local target="$1"
+
+    if [ -z "$target" ]; then
+        echo "repack-kernel: missing target for pack"
+        exit 1
+    fi
+    snap pack "${@:2}" "$target/kernel" 
+}
+
+main() {
+    if [ $# -eq 0 ]; then
+        show_help
+        exit 0
+    fi
+
+    local subcommand="$1"
+    local action=
+    while [ $# -gt 0 ]; do
+        case "$1" in
+            -h|--help)
+                show_help
+                exit 0
+                ;;
+            *)
+                action=$(echo "$subcommand" | tr '-' '_')
+                shift
+                break
+                ;;
+        esac
+    done
+
+    if [ -z "$(declare -f "$action")" ]; then
+        echo "repack-kernel: no such command: $subcommand" >&2
+        show_help
+        exit 1
+    fi
+
+    "$action" "$@"
+}
+
+main "$@"
```

Comment:
new line please

Reply:
if you open the script locally there is a newline after main "$@", you just can't see it in online editor as github does not show the last newline